common/djangoapps/third_party_auth/api/views.py

```diff
@@ -20,13 +24,129 @@
+        OAuth2AuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    throttle_classes = [ProviderSustainedThrottle, ProviderBurstThrottle]
+
+    def do_get(self, request, identifier):
+        """
+        Fulfill the request, now that the identifier has been specified.
+        """
+        is_unprivileged = self.is_unprivileged_query(request, identifier)
+
+        if is_unprivileged:
+            if not getattr(settings, 'ALLOW_UNPRIVILEGED_SSO_PROVIDER_QUERY', False):
+                return Response(status=status.HTTP_403_FORBIDDEN)
+        try:
+            user = User.objects.get(**{identifier.kind: identifier.value})
+        except User.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        providers = pipeline.get_provider_user_states(user)
+
+        active_providers = [
+            self.get_provider_data(assoc, is_unprivileged)
+            for assoc in providers if assoc.has_account
+        ]
+
+        # In the future this can be trivially modified to return the inactive/disconnected providers as well.
+
+        return Response({
+            "active": active_providers
+        })
+
+    def get_provider_data(self, assoc, is_unprivileged):
+        """
+        Return the data for the specified provider.
+
+        If the request is unprivileged, do not return the remote ID of the user.
+        """
+        provider_data = {
+            "provider_id": assoc.provider.provider_id,
+            "name": assoc.provider.name,
+        }
+        if not is_unprivileged:
+            provider_data["remote_id"] = assoc.remote_id
+        return provider_data
+
+    def is_unprivileged_query(self, request, identifier):
+        """
+        Return True if a non-superuser requests information about another user.
+
+        Params must be a dict that includes only one of 'username' or 'email'
+        """
+        assert identifier.kind in {'username', 'email'}
+
+        self_request = False
+        if identifier == self.identifier('username', request.user.username):
+            self_request = True
+        elif identifier.kind == 'email' and getattr(identifier, 'value', object()) == request.user.email:
+            # AnonymousUser does not have an email attribute, so fall back to
+            # something that will never compare equal to the provided email.
+            self_request = True
+        if self_request:
+            # We can always ask for our own provider
+            return False
+        # We are querying permissions for a user other than the current user.
+        if not request.user.is_superuser and not ApiKeyHeaderPermission().has_permission(request, self):
+            # The user does not have elevated permissions.
+            return True
+        return False
+
+
+class UserView(BaseUserView):
     """
     List the third party auth accounts linked to the specified user account.
 
+    [DEPRECATED]
```

Comment:
nit: add a sentence about renaming UserView2 to UserView after ripping this one out.

Reply:
Better yet, I'll rename now, and call the old one UserViewDeprecated

Reply:
On second thought, there could be extension libraries that are referring to this.  Safer just to add the note.